### PR TITLE
Rework header according to UI mockups

### LINKED ISSUE #55: Rework Header according to UI Mockups
Rework the header according to the [this](https://www.figma.com/file/G7m9jZRwr2vtD1ZTX9iKd5/Patient-Dashboard-UI-Wireframes?node-id=0%3A1) UI Mockup. 

Please remove the sidebar component from the desktop while your're at it, as we are going for a full mobile first approach in the patient dashboard now.

### PR DESCRIPTION
**Pull request by ValiHar**
Header reworked according to UI mockups and fixed bugs regarding stacking contexts with the dividers.

### CODE REVIEW

**Review by n-krueger — CHANGES_REQUESTED** [review]
Please remove unused vars that as indicated by ESLint:
```
WARNING in src/components/dashboard/appShell/header.tsx
  Line 45:10:  'profile' is assigned a value but never used   @typescript-eslint/no-unused-vars
  Line 50:9:   'navigate' is assigned a value but never used  @typescript-eslint/no-unused-vars

src/views/Wiki/vaccineWiki.tsx
  Line 6:3:  'IconButton' is defined but never used  @typescript-eslint/no-unused-vars

```

Code LGTM.

One note on the design of the header: It reminds me of search boxes in other design systems, which communicates the affordance that users may enter text in it. For us it is only a container for other UI elements. This is probably a holdover from the original designs which did include a search box but we should see if we can make it's design more inline with what we want the user to expect.

File: src/components/dashboard/appShell/header.tsx [hunk 1]
```diff
@@ -3,122 +3,103 @@ import {
...
+
+const LinkItems: Array<LinkItemProps> = [
+  { name: 'History', icon: FaSyringe, link: 'history' },
+  { name: 'Recommendations', icon: FaRegCalendarAlt, link: 'recommendations' },
+  { name: 'Wiki', icon: FaBookMedical, link: 'wiki' },
+];
+
+export const DashboardHeader: FC<DashboardProps> = ({}) => {
   const { keycloak } = useKeycloak();
   const [profile, setProfile] = useState<KeycloakProfile | undefined>();
```
Comment (n-krueger) [inline src/components/dashboard/appShell/header.tsx:42]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const [profile, setProfile] = useState<KeycloakProfile | undefined>();
```
@FuchsDominik use this if you want to keep this code around for later use.

Comment (FuchsDominik) [reply src/components/dashboard/appShell/header.tsx:42]: Good idea, this way we solve the linting error

**Review by n-krueger — APPROVED** [review]
LGTM!

**Pull request merged**